src/lib/PetitioRequest.ts

```diff
@@ -129,6 +134,18 @@ export class PetitioRequest {
 
 		return this;
 	}
+	/**
+	 * @param {AbortController} controller A controller instance that handles aborting the request.
+	 * @return {*} The request object for further composition.
+	 * @example Initiate a AbortController class and then attach it to the request object
+	 * by doing `.signal(the_controller)`. Then you can abort the request by running
+	 * the abort() function on the controller you created.
```

Comment:
```suggestion
	 * @example
	 * ```ts
	 * const controller = new AbortController();
	 * const result = petitio(URL).signal(controller);
	 * setTimeout(() => controller.abort(), 5000) // serves as a request timeout
	 * ```
	 */
```